apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/EnumMap.cs

```diff
@@ -71,6 +76,35 @@ internal static string ToApiValue(T value, string paramName = "value")
             throw new ArgumentException($"Value {value} is undefined in {typeof(T).Name}", paramName);
         }
 
+        /// <summary>
+        /// Similar to <see cref="ToApiValue(T, string)"/> but for used with Flags enum.
+        /// </summary>
+        /// <param name="value">An enum value, possibly flagged.</param>
+        /// <returns>A set of string that contains the api value corresponding to each
+        /// of the flags set in <paramref name="value"/>.</returns>
+        internal static ISet<string> ToApiValues(T value)
+        {
+            int intValue = (int)(object) value;
+            ISet<string> apiValues = new HashSet<string>();
+            foreach (int oneValue in s_intValueToString.Keys)
```

Comment:
If you iterate over the key value pairs instead of just the keys, you don't need the lookup in the Add call.